cytoolz/itertoolz.pyx

```diff
@@ -1052,7 +1056,13 @@ cdef class partition_all:
         # iterable exhausted before filling the tuple
         if i == 0:
             raise StopIteration
-        return PyTuple_GetSlice(result, 0, i)
+        # fill in the rest of the tuple
+        j = i
+        while i < self.n:
+            Py_INCREF(none)
+            PyTuple_SET_ITEM(result, i, none)
```

Comment:
`result` never makes it to Python, I am not sure how these nulls cause a problem.